Enigmatry.Entry.Blueprint.Infrastructure/Identity/PrincipalProviderCurrentUserProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Enigmatry.Entry.Blueprint.Domain.Identity;
+using Enigmatry.Entry.Blueprint.Domain.Users;
+using Enigmatry.Entry.Core.Data;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Enigmatry.Entry.Blueprint.Infrastructure.Identity;
+
+[UsedImplicitly]
+public class PrincipalProviderCurrentUserProvider(
```

Comment:
Renamed class CurrentUserProvider

Reply:
This was done to make explicit that this implementation has a dependency in IPrincipal (i.e. it requires authentication and claims to be setup, which makes it more difficult to use in the scheduler or tests).  

This is the fix that removed the workaround in the PopulateCreatedUpdateInterceptor.cs. PopulateCreatedUpdateInterceptor used to check whether UserId was not null and exit early in that situation leaving CreatedById and UpdatedById to have Guid.Empty which caused problems in the tests seeding.
 
Originally, we have had three implementations of ICurrentUserProvider: 
- CurrentUserProvider (that has dependency on IClaimsProvider which has dependency on IPrincipal and ClaimsPrincipal (ClaimsPrincipal cannot be resolve without setting up the Authentication system)
- NullCurrentUserProvider - the class is no longer being used so it was removed
- SystemUserProvider - had a hard-code UserId
In our tests we are seeding test user (under which the tests will be executed in the api) but none of the implementations allow id of the user to be change to test user. 
The fix to remove the workaround was to introduce TestUserProvider (which inherits from SystemUserProvider but overrides the userId). During that fix, I removed NullCurrentUserProvider and renamed CurrentUserProvider to make its dependency more explicit. 

If you think that it is better to rename it back to CurrentUserProvider that is also fine. I don't have any strong preference here.

Reply:
This is what is seen in the tools:
![image](https://github.com/user-attachments/assets/f189b588-2d96-4e72-a430-cf8a64de39a9)

Reply:
My idea was to make this more similar to the "System" and "Test"

Reply:
Yeah, I understand your reason. It's just that most devs are used to just 'CurrentUserProvder'. And from a naming perspective... it does provide the current user, while SystemUserProvider provides a system user.

Reply:
TestUserProvider could IMO also be TestCurrentUserProvider. Both make sense.

Reply:
Ok, your point also makes sense, I will then consolidate into CurrentUserProvider, SystemUserProvider and TestUserProvider